src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -812,101 +731,204 @@ public int getCurrentRunningCount() {
+      boolean queued = false;
+      try {
+        idle: while (true) {
+          int casPoolSize;
+          if (shutdownFinishing) {
+            currentPoolSize.decrementAndGet();
+            worker.stopIfRunning();
+            return null;
+          } else if ((casPoolSize = currentPoolSize.get()) > maxPoolSize) {
+            if (currentPoolSize.compareAndSet(casPoolSize, casPoolSize - 1)) {
+              worker.stopIfRunning();
+              return null;
+            }
+          } else {
+            // ready to poll task, we loop here to avoid pool state recheck logic unless we have to block
+            while (true) {
+              TaskWrapper nextTask = queueManager.getNextTask();
+              if (nextTask == null) {
+                if (queued) {
+                  // we can only park after we have queued, then checked again for a result
+                  LockSupport.park();
+                  continue idle;
+                } else {
+                  addWorkerToIdleChain(worker);
+                  queued = true;
+                }
+              } else {
+                /* TODO - right now this has a a deficiency where a recurring period task can cut in 
+                 * the queue line.  The condition would be as follows:
+                 * 
+                 * * Thread 1 gets task to run...task is behind execution schedule, likely due to large queue
+                 * * Thread 2 gets same task
+                 * * Thread 1 gets reference, executes, task execution completes
+                 * * Thread 2 now gets the reference, and execution check and time check pass fine
+                 * * End result is that task has executed twice (on expected schedule), the second 
+                 *     execution was unfair since it was done without respects to queue order and 
+                 *     other tasks which are also likely behind execution schedule in this example
+                 *     
+                 * This should be very rare, but is possible.  The only way I see to solve this right 
+                 * now is to introduce locking.
+                 */
+                // must get executeReference before time is checked
+                short executeReference = nextTask.getExecuteReference();
+                long taskDelay = nextTask.getScheduleDelay();
+                if (taskDelay > 0) {
+                  if (taskDelay == Long.MAX_VALUE) {
+                    // TODO - this will cause a spin if there is only one recurring task
+                    // don't sleep forever, this task is just starting to execute, so spin
+                    Thread.yield();
+                  }
+                  if (queued) {
+                    if (! workerTimedPark) {
+                      // we can only park after we have queued, then checked again for a result
+                      workerTimedPark = true;
+                      LockSupport.parkNanos(Clock.NANOS_IN_MILLISECOND * taskDelay);
```

Comment:
It seems like more then one thread could end up in here at a time.  Or does queued some how take care of that?